### ISSUE TITLE
Unable to create an InstanceTemplate with the namePrefix property set

### ISSUE THREAD

**Issue by JONBRWN** [pulumi/pulumi-gcp#227]
The following code block is provided in the `InstanceTempate.ts` file:

```
const instanceTemplate = new gcp.compute.InstanceTemplate("instanceTemplate", {
    // boot disk
    disks: [{
        sourceImage: "debian-cloud/debian-9",
    }],
    machineType: "n1-standard-1",
    namePrefix: "instance-template-",
    region: "us-central1",
});
```

but fails on creation with the following error:

```
Diagnostics:
  gcp:compute:InstanceTemplate (instanceTemplate):
    error: gcp:compute/instanceTemplate:InstanceTemplate resource 'instanceTemplate' has a problem: "name": conflicts with name_prefix
```

The Terraform documentation states that the `name` and `name_prefix` arguments are optional but it appears that the `name` argument is mandatory on the Pulumi side.

**Comment by clstokes** [comment]
Seems to be a similar/related issue to https://github.com/pulumi/pulumi-aws/issues/143.

**Comment by stack72** [comment]
This was because `name_prefix` wasn't marked as conflicting with `name`.

https://github.com/terraform-providers/terraform-provider-google-beta/pull/1202

**Comment by ndmckinley** [comment]
For what it's worth, `name` _is_ marked as conflicting with `name_prefix`, which provides the same experience on the command line - consider changing your tool to understand that dependency in both directions.

**Comment by stack72** [comment]
Ok, this will be included in the v1.3.0 release that is pending :)

**Comment by schmurfy** [comment]
I just got this trying out pulumi, I am using this file straight from the [documentation](https://www.pulumi.com/docs/reference/pkg/gcp/compute/instancetemplate/) (with update to match the new syntax):
```typescript
import * as gcp from "@pulumi/gcp";

const instanceTemplate = new gcp.compute.InstanceTemplate("instanceTemplate", {
    namePrefix: "instance-template-",
    machineType: "e2-medium",
    region: "us-central1",
    disks: [{}],
    networkInterfaces: [{}],
});
const instanceGroupManager = new gcp.compute.InstanceGroupManager("instanceGroupManager", {
    versions: [
      {
        name: "primary",
        instanceTemplate: instanceTemplate.id,
      }
    ],
    
    baseInstanceName: "instance-group-manager",
    zone: "us-central1-f",
    targetSize: 1,
});
```

and I get this:
```
Diagnostics:
  gcp:compute:InstanceTemplate (instanceTemplate):
    error: gcp:compute/instanceTemplate:InstanceTemplate resource 'instanceTemplate' has a problem: ConflictsWith: "name": conflicts with name_prefix. Examine values at 'InstanceTemplate.Name'.
```

I thought I was using an old version but I am using pulumi `3.9.1` and the latest `pulumi-gcp` package:
```
{
    "name": "test-project",
    "devDependencies": {
        "@types/node": "^10.0.0"
    },
    "dependencies": {
        "@pulumi/gcp": "^5.14.0",
        "@pulumi/pulumi": "^3.0.0"
    }
}
```
